pkg/port/iscsit/iscsid.go

```diff
@@ -873,6 +880,10 @@ func (s *ISCSITargetDriver) scsiCommandHandler(conn *iscsiConnection) (err error
 			conn.session.PendingTasksMutex.Unlock()
 		}
 	case OpNoopOut:
+		timeout := time.Since(conn.lastNopout)
+		if timeout > opPingTimeout {
+			log.Warningf("Received ping after %v, ping timeout is %v", timeout, opPingTimeout)
+		}
 		iscsiExecNoopOut(conn)
```

Comment:
we need to check the time taken by iscsiExecNoopOut function call. We don't need to check the time from last nopout.
Its a good idea to track it, but, doing this here won't help and the same need to be done in a long-time running thread. Lets take this in another PR.
For this PR, just find the time taken to execute 'iscsiExecNoopOut'. And, do a warning log if it takes more than 3 seconds

Reply:
ok